pkg/kubernetes/deploy.js

```diff
@@ -77,6 +81,192 @@ define([
+                var label = prefix + "-" + param.param_key + "-" + lsuffix;
+                var input = prefix + "-" + param.param_key + "-" + isuffix;
+                var tmp = { "label_val" : label, "input_val" : input };
+                labels.push(tmp);
+            }
+        }
+        return labels;
+    }
+
+
+    /*
+     * Get app_name and KEY from label
+     */
+    function get_app_from_label(label) {
+        var app_name = "";
+        var key = label.split("-label")[0].split("-");
+        key = key[key.length-1];
+        app_name = label.split("-label")[0].split("-" + key)[0];
+        return { "app_name" : app_name, "key" : key };
+    }
+
+    /*
+     * Convert default ini file to Key-Value format
+     * to be used in creating the form
+     */
+    function convert_to_KV(ans) {
+        var applist = [];
+
+        for(var key in ans) {
+            var app = { "app_name" : "", "app_params" : []};
+            // Ignore general
+            if (key === "general")
+                continue;
+
+            app.app_name = key;
+            for(var x in ans[key]) {
+                var tmpKV = { "param_key" : "", "param_value": ""};
+                tmpKV.param_key = x;
+
+                if(ans[key][x])
+                    tmpKV.param_value = ans[key][x];
+                else
+                    tmpKV.param_value = "";
+
+                app.app_params.push(tmpKV);
+            }
+            applist.push(app);
+
+        }
+        return applist;
+    }
+
+    function validate_app_fields() {
+        var dfd = $.Deferred();
+        var ex, fails = [];
+        var answer_data = nulecule_client.answers;
+        var label_input = get_fields(convert_to_KV(answer_data));
+        var fields = { };
+
+
+        for(var x in label_input) {
+            var li = label_input[x];
+            var input_id = "#" + li.input_val;
+            var label_id = li.label_val;
+            var app_data = get_app_from_label(label_id);
+            var input_value = $(input_id).val();
+
+            answer_data[app_data.app_name][app_data.key] = input_value;
+
+            var label_value = $('#deploy-app-dialog').find('label[for="' + label_id + '"]').text().trim();
+            if (!input_value)
+                ex = new Error(label_value + " cannot be empty.");
+
+            if (ex) {
+                ex.target = input_id;
+                fails.push(ex);
+                ex = null;
+            }
+        }
+
+        var ns = $("#deploy-app-namespace").val();
+        if (!ns)
+            ex = new Error(_("Namespace cannot be empty."));
+        else if (!/^[a-z0-9]+$/i.test(ns))
+            ex = new Error(_("Please provide a valid namespace."));
+        if (ex) {
+            ex.target = "#deploy-app-namespace-group";
+            fails.push(ex);
+            ex = null;
+        } else {
+            answer_data.general.namespace = ns;
+        }
+
+        var nulecule_image = $("#deploy-app-nulecule-image");
```

Comment:
No, they should not be changed.
I have added code to disable them after install step.